### PR TITLE
[WIP] Awesome DHT

### PR DESCRIPTION
**Pull request by dignifiedquire**
Work tracker: https://github.com/libp2p/js-libp2p-dht/pull/1

- [x] update API docs for peerRouting and contentRouting https://hackmd.io/JwVgTGDGAmLAtAIwCyQMz2QBnEr1F4B2LADhAEMBGANkmGRAFMg=?both

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -117,22 +123,29 @@ class Node extends EventEmitter {
...
       if (err) {
         return callback(err)
       }
+
       if (ws) {
-        this.swarm.transport.add(ws.tag || ws.constructor.name, ws)
+        this.swarm.transport.add(
+          ws.tag || ws.constructor.name, ws
+        )
       }
```
Comment (diasdavid) [inline src/index.js]: transports (swarm) prep needs to happen first than discovery as discovery will trigger dials that use swarm.

File: README.md [hunk 2]
```diff
@@ -144,6 +147,36 @@ class Node extends libp2p {
...
+
+#### `libp2p.contentRouting.provide(key, timeout, callback)`
+
+- `key`: 
+- `timeout`: Number miliseconds
+
+#### `libp2p.dht.put(key, value, callback)`
+
+- `key`: 
+- `value`:
```
Comment (diasdavid) [inline README.md]: @dignifiedquire can you clarify here what type can be key and value? If it is "String" and "Buffer" cause the implementation lets you to, let's instead direct the users to the things we want them to actually use the DHT.

Comment (dignifiedquire) [inline README.md]: BUFFER

Comment (diasdavid) [inline README.md]: - [ ] Update docs to say `Buffer`

File: README.md [hunk 3]
```diff
@@ -95,7 +96,9 @@ class Node extends libp2p {
       },
       discovery: [
         new MulticastDNS(peerInfo, 'your-identifier')
-      ]
+      ],
+      // DHT is passed as its own enabling PeerRouting, ContentRouting and DHT itself components
+      dht: new DHT()
```
Comment (dignifiedquire) [inline README.md]: missing arguments

**Pull request merged**